frontend/src/metabase/entities/collections.js

```diff
@@ -324,3 +325,36 @@ function byCollectionUrlId(state, { params, location }) {
 function byCollectionQueryParameter(state, { location }) {
   return location && location.query && location.query.collectionId;
 }
+
+function hasIntersection(list1, list2) {
+  if (!list2) {
+    return false;
+  }
+  return _.intersection(list1, list2).length > 0;
+}
+
+export function buildCollectionTree(collections, { targetModels } = {}) {
```

Comment:
I suppose I can see why you prefer this function here.

At the same time, this file is over 300 loc as it is.

I wonder if you'd consider importing/reexporting such a function from this very file, and inaugurate a directory for collections entity.

So move this file to something like

`frontend/src/metabase/entities/collections/index.js`

and we place this function and the above in something like

`frontend/src/metabase/entities/collections/tree.js`

Many ways to organize this, the principle being to keep file line count not much over 150 loc or so if we can.